drop-in/src/main/java/com/adyen/checkout/dropin/ui/action/ActionComponentDialogFragment.kt

```diff
@@ -197,10 +154,24 @@ class ActionComponentDialogFragment : DropInBottomSheetDialogFragment(), Observe
         protocol.showError(getString(R.string.action_failed), componentError.errorMessage, true)
     }
 
+    // TODO this should rely on the generic component not the individual providers
     private fun shouldFinishWithAction(): Boolean {
         return getActionProviderFor(action)?.providesDetails() == false
     }
 
+    fun handleRedirectResponse(intent: Intent) {
+        handleIntent(intent)
+    }
+
+    fun handleWeChatPayResponse(intent: Intent) {
+        handleIntent(intent)
+    }
+
+    private fun handleIntent(intent: Intent) {
```

Comment:
If we make this public, then we don't need the `handleRedirectResponse` and `handleWeChatPayResponse` methods